library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types.hs

```diff
@@ -101,6 +103,31 @@ data Errors
     -- | typechecker, if the type of the term did not come directly from a
     -- | test, or some other module that bypasses the typechecker)
     MismatchedFieldTypes Text
+  | -- | Found a sum term with an undeclared sum type (this indicates a
+    -- | bug in the typechecker, if the type of the term did not come
+    -- | directly from a test, or some other module that bypasses the
+    -- | typechecker)
+    NonExistentSumType Text
+  | -- | Found a sum term whose variants' types did not match the variant
+    -- | types in the sum type declaration (this indicates a bug in the
+    -- | typechecker, if the type of the term did not come directly from a
+    -- | test, or some other module that bypasses the typechecker)
+    MismatchedVariantTypes Text
+  | -- | Found a variant term with a name not present in its type's list of
+    -- | variants (this indicates a bug in the typechecker, if the type of the
+    -- | term did not come directly from a test, or some other module that
+    -- | bypasses the typechecker)
+    NonExistentVariant Text
+  | -- | Found a match term whose cases' types did not match the variant
+    -- | types in the sum type declaration (this indicates a bug in the
+    -- | typechecker, if the type of the term did not come directly from a
+    -- | test, or some other module that bypasses the typechecker)
+    MismatchedCaseTypes Text
+  | -- | Found a sum term whose type did not match the one found by
+    -- | looking up the type's name (this indicates a bug in the
+    -- | typechecker, if the type of the term did not come directly from a
+    -- | test, or some other module that bypasses the typechecker)
+    MismatchedSumTypes Text
```

Comment:
Nice use of comments, I wonder if | is needed after every --? It seems like it was like that before, but I wonder what haddock wants.

Reply:
only the first `|` is needed by haddock. I think @rokopt has used `|` in every line so it looks more uniform.

Reply:
Uniformity is good, though I think since we o gen out haock, oes this affect it at all?

Reply:
no, this does not affect the output. I'm fine with or without the extra pipes. However, not having them is slightly more convinient for me since when I do `M-q` in emacs to format the haddock paragraph, it does not add the extra pipes.